packages/dds/merge-tree/src/test/PartialLengths.perf.spec.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { benchmark, BenchmarkType } from "@fluid-tools/benchmark";
+import { MergeTreeDeltaType } from "../ops";
+import { MergeTree } from "../mergeTree";
+import { insertText, markRangeRemoved } from "./testUtils";
+
+describe("partial lengths perf", () => {
+	for (const incremental of [true, false]) {
+		benchmark({
+			type: BenchmarkType.Measurement,
+			title: `incremental updates = ${incremental}`,
+			category: "partial lengths",
+			before: () => {
+				MergeTree.options.incrementalUpdate = incremental;
+			},
+			benchmarkFn: async () => {
+				const mergeTree = new MergeTree();
+
+				let i = 1;
+				for (; i < 1001; i++) {
+					insertText({
+						mergeTree,
+						pos: 0,
+						refSeq: i,
+						clientId: 0,
+						seq: i,
+						text: "a",
+						props: undefined,
+						opArgs: { op: { type: MergeTreeDeltaType.INSERT } },
+					});
+				}
+
+				for (; i < 2001; i++) {
+					markRangeRemoved({
+						mergeTree,
+						start: i - 1001,
+						end: i - 1000,
+						refSeq: i,
+						clientId: 0,
+						seq: i,
+						opArgs: { op: { type: MergeTreeDeltaType.REMOVE } },
+						overwrite: false,
+					});
+				}
+
+				for (; i < 3001; i++) {
+					insertText({
+						mergeTree,
+						pos: 0,
+						refSeq: i,
+						clientId: 0,
+						seq: i,
+						text: "a",
+						props: undefined,
+						opArgs: { op: { type: MergeTreeDeltaType.INSERT } },
+					});
+				}
+			},
+			after: () => {
+				MergeTree.options.incrementalUpdate = true;
```

Comment:
nit: probably better to store the original value and restore it rather than turn it on. don't feel super strongly, as it's unlikely we'd ever change this default.